test/box-luatest/lua_memory_test.lua

```diff
@@ -0,0 +1,68 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group('lua_memory setting test')
+--local treegen = require('test.treegen')
+
+local MB = 1024 * 1024
+local LUA_MIN_MEMORY = 256 * MB
+
+g.before_each(function()
+    g.server = server:new{
+        alias = 'default',
+        box_cfg = {lua_memory = LUA_MIN_MEMORY}
+    }
+    g.server:start()
+
+end)
+
+g.after_each(function()
+    g.server:drop()
+end)
+
+g.test_lua_memory_setting = function(g)
+    g.server:exec(function()
+        local MB = 1024 * 1024
+        local LUA_MIN_MEMORY = 256 * MB
+        local LUA_EXTENDED_MEMORY = 600 * MB
+
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY)
+
+        -- Increasing the limit should be ok.
+        box.cfg{lua_memory = LUA_MIN_MEMORY + 1}
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY + 1)
+
+        local err_min = 'Cannot limit the Lua memory with values less than '..
+                        '256MB'
+        -- Setting the limit less than 256MB is prohibited.
+        t.assert_error_msg_equals(err_min, box.cfg,
+            {lua_memory = LUA_MIN_MEMORY - 1})
+
+        box.cfg{lua_memory = LUA_EXTENDED_MEMORY}
+
+        local used = collectgarbage('count') * 1024
+        local free = LUA_EXTENDED_MEMORY - used
+        local _ = string.rep('a', free / 3)
```

Comment:
Please add a comment that this string should be successfully allocated.

Reply:
The comment was added.

Reply:
Nit: Do you mean `1/3`?

Reply:
Actually, it allocates ~2/3 of the free memory. Last assignment allocates `free / 3 * 2` memory:
```
tarantool> _ = string.rep('1', 1000 * 1000 * 1000)
freed 6144
allocated 1073733632
allocated 1000000025
```
These log was got by adding printing allocations/frees with size >=4kb.

Some experiments made me think, that there is some kind of a cache for such large calls. The memory is allocated for `string.rep()` call result with the specified parameters and for the variable itself separately. It's easy to see because the later calls `any_other_var = string.rep('1', 1000 * 1000 * 1000)` won't cause any additional large allocations.

Reply:
I suppose this is due to the growth of the string buffer. Let's drop the mention of the specific memory amount to avoid confusion.

Side note: Later calls don't cause allocations since LuaJIT uses string interning, so there is no additional string created.

Reply:
Dropped the mention of the strange `2/3` value. Thanks for the side note.